mmengine/optim/optimizer/zero_optimizer.py

```diff
@@ -53,6 +57,15 @@ def __init__(self, params, optimizer_type: str, **kwargs):
             '`torch.distributed.optim.ZeroReundancyOptimizer` is only '
             'available when pytorch version >= 1.8.')
         assert is_available(), 'torch.distributed.rpc is not available.'
+        # Avoid the generator becoming empty after the following check
+        params = list(params)
```

Comment:
This line can be removed.

Reply:
The comment seems to say "`params` become empty when `params`'s type is a generator after the assert line".

Reply:
got it